backend/zoho/zoho.go

```diff
@@ -0,0 +1,1223 @@
+		return nil, errors.New("couldn't copy file: invalid response")
+	}
+	// Only set ID here because response is not complete Item struct
+	dstObject.id = result.Items[0].ID
+
+	// Can't copy and change name in one step so we have to check if we have
+	// the correct name after copy
+	if f.opt.Enc.ToStandardName(result.Items[0].Attributes.Name) != leaf {
+		if err = dstObject.rename(ctx, leaf); err != nil {
+			return nil, errors.Wrap(err, "copy: couldn't rename copied file")
+		}
+	}
+	return dstObject, nil
+}
+
+func (f *Fs) move(ctx context.Context, srcID, parentID string) (item *api.Item, err error) {
+	// Move the object
+	opts := rest.Opts{
+		Method:       "PATCH",
+		Path:         "/files",
+		ExtraHeaders: map[string]string{"Accept": "application/vnd.api+json"},
+	}
+	moveFile := api.WriteMultiMetadataRequest{
+		Meta: []api.WriteMetadata{
+			{
+				Attributes: api.WriteAttributes{
+					ParentID: parentID,
+				},
+				ID:   srcID,
+				Type: "files",
+			},
+		},
+	}
+	var resp *http.Response
+	var result *api.ItemList
+	err = f.pacer.Call(func() (bool, error) {
+		resp, err = f.srv.CallJSON(ctx, &opts, &moveFile, &result)
+		return shouldRetry(resp, err)
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "move failed")
+	}
+	// Server acts weird some times make sure our array actually contains
+	// a file
+	if len(result.Items) != 1 {
+		return nil, errors.New("move failed: invalid response")
+	}
+	return &result.Items[0], nil
+}
+
+// Move src to this remote using server side move operations.
+//
+// This is stored with the remote path given
+//
+// It returns the destination Object and a possible error
+//
+// Will only be called if src.Fs().Name() == f.Name()
+//
+// If it isn't possible then return fs.ErrorCantMove
+func (f *Fs) Move(ctx context.Context, src fs.Object, remote string) (fs.Object, error) {
+	srcObj, ok := src.(*Object)
+	if !ok {
+		fs.Debugf(src, "Can't move - not same remote type")
+		return nil, fs.ErrorCantMove
+	}
+	err := srcObj.readMetaData(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	srcDir, _ := srcObj.fs.splitPath(src.Remote())
+	dstDir, dstFilename := f.splitPath(remote)
+	// same parent only need to rename
+	if srcDir == dstDir {
+		if err = srcObj.rename(ctx, dstFilename); err != nil {
+			return nil, err
+		}
+		return srcObj, nil
+	}
+
+	// Create temporary object
+	dstObject, _, directoryID, err := f.createObject(ctx, remote, srcObj.size, srcObj.modTime)
+	if err != nil {
+		return nil, err
+	}
+	item, err := f.move(ctx, srcObj.id, directoryID)
+	if err != nil {
+		return nil, err
+	}
+	// Only set ID here because response is not complete Item struct
+	dstObject.id = item.ID
+
+	// Can't move and change name in one step so we have to check if we have
+	// the correct name after move
+	if f.opt.Enc.ToStandardName(item.Attributes.Name) != dstFilename {
+		if err = dstObject.rename(ctx, dstFilename); err != nil {
+			return nil, errors.Wrap(err, "move: couldn't rename moved file")
+		}
+	}
```

Comment:
Zoho cannot move and rename an object or directory in a single API call. Renaming must always be done separately. I've decided to go with move first and rename afterward.

Reply:
There is a little dance you need to do here. I see from the code that you are doing that already! Eg

https://github.com/rclone/rclone/blob/fc5b14b62046b240efd5963bb62a29ae47c0a747/backend/sharefile/sharefile.go#L921-L926

Reply:
Ah true I should be using a different name during the move, didn't think of that.